src/tests/system/tests/test_gpo.py

```diff
@@ -981,3 +978,46 @@ def test_gpo__works_when_auto_private_groups_is_set_true(client: Client, ad: AD,
     result = client.tools.id("user1")
     assert result is not None, "User not found!"
     assert result.memberof("domain users"), "User missing from group 'domain users'!"
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.parametrize("method", ["ssh", "su"])
+@pytest.mark.topology(KnownTopology.AD)
+@pytest.mark.ticket(gh=7452)
+def test_gpo__works_when_auto_private_group_is_true_using_posix_accounts(client: Client, ad: AD, method: str):
+    """
+    :title: GPO evaluation fails when auto_private_groups is set to true and ldap_id_mapping is disabled
+    :setup:
+        1. Create the following user 'user1' and 'deny_user1' with uids and gids
+        2. Create and link the GPO 'site policy' and add 'user1' and 'Domain Admins' to
+           SeInteractiveLogonRight key. Add 'deny_user1 to SeDenyInteractiveLogonRight key'
+        3. Configure sssd.conf with 'ad_gpo_access_control = enforcing', 'auto_private_groups = true' and
+           'ldap_id_mapping = false'
+        4. Start SSSD
+    :steps:
+        1. Authenticate as 'user1'
+        2. Authenticate as 'deny_user1'
+    :expectedresults:
+        1. Authentication is successful
+        2. Authenticated user is unsuccessful
+    :customerscenario: True
+    """
+    user1 = ad.user("user1").add(uid=10000, gid=10000)
+    deny_user1 = ad.user("deny_user1").add(uid=10001, gid=10001)
+
+    ad.gpo("site policy").add().policy(
+        {
+            "SeInteractiveLogonRight": [user1, ad.group("Domain Admins")],
+            "SeDenyInteractiveLogonRight": [deny_user1],
+        }
+    ).link()
+
+    client.sssd.domain["ad_gpo_access_control"] = "enforcing"
+    client.sssd.domain["auto_private_groups"] = "true"
+    client.sssd.domain["ldap_id_mapping"] = "false"
+    client.sssd.start()
+
+    assert client.auth.parametrize(method).password(
+        "user1", password="Secret123"
+    ), "Allowed user authentication failed!"
+    assert not client.auth.parametrize(method).password("deny_user1", password="Secret123"), "Denied user logged in!"
```

Comment:
Hi,

it looks like you are consolidating on multi-line assert statements. Shouldn't this one be on multiple lines as well?

bye,
Sumit

Reply:
@sumit-bose black just keeps on reformatting it this way.

Reply:
Hi,

thanks, given the assert before I wonder what the rule for line-breaking is here. Maybe it does not break the last line?

bye,
Sumit